llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -2666,15 +2680,21 @@ static bool isKnownNonZeroFromOperator(const Operator *I,
 
       // The condition of the select dominates the true/false arm. Check if the
       // condition implies that a given arm is non-zero.
-      Value *X;
+      Value *X, *Y;
       CmpInst::Predicate Pred;
-      if (!match(I->getOperand(0), m_c_ICmp(Pred, m_Specific(Op), m_Value(X))))
+      if (!match(I->getOperand(0), m_ICmp(Pred, m_Value(X), m_Value(Y))))
```

Comment:
It seems like a NFC change. Does this change improve matching performance?

Reply:
Its necessary now that we can use non-constants. I'll seperate that to an NFC and rebase this.

Reply:
Will probably abandon this one per nikic's comments, but will update the matching logics